src/main/python/measure_judged.py

```diff
@@ -0,0 +1,69 @@
+"""Script to measure the percentage of judged documents at various
+cutoffs."""
+import argparse
+import collections
+
+from typing import Dict
+from typing import List
+from typing import Set
+
+
+def load_qrels(path: str) -> Dict[str, Set[str]]:
+    """Loads qrels into a dict of key: query_id, value: set of relevant doc
+    ids."""
+    qrels = collections.defaultdict(set)
+    with open(path) as f:
+        for i, line in enumerate(f):
+            line = ' '.join(line.split())
+            query_id, _, doc_id, relevance = line.rstrip().split(' ')
+            qrels[query_id].add(doc_id)
+            if i % 1000000 == 0:
+                print('Loading qrels {}'.format(i))
+    return qrels
+
+
+def load_run(path: str) -> Dict[str, List[str]]:
+    """Loads run into a dict of key: query_id, value: list of candidate doc
+    ids."""
+    run = collections.OrderedDict()
+    with open(path) as f:
+        for line in f:
+            query_id, _, doc_title, rank, _, _ = line.split(' ')
+            if query_id not in run:
+                run[query_id] = []
+            run[query_id].append((doc_title, int(rank)))
+
+    # Sort candidate docs by rank.
+    sorted_run = collections.OrderedDict()
+    for query_id, doc_titles_ranks in run.items():
+        sorted(doc_titles_ranks, key=lambda x: x[1])
+        doc_titles = [doc_titles for doc_titles, _ in doc_titles_ranks]
+        sorted_run[query_id] = doc_titles
+
+    return sorted_run
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description='measure the percentage of judged documents at various '
+                    'cutoffs.')
+    parser.add_argument('--qrels', type=str, required=True, help='qrels file')
+    parser.add_argument('--run', type=str, required=True, help='run file')
+
+    args = parser.parse_args()
+
+    qrels = load_qrels(args.qrels)
+    run = load_run(args.run)
+
+    for max_rank in [5, 10, 20, 30]:
```

Comment:
maybe take this in as a command-line arg, e.g., `5,10,20,30` with this being a default?

Reply:
Good idea!